### ISSUE TITLE
Is it possible to run BAGEL with PyTorch 2.3.1 and FlashAttention 2.5.8 on CUDA 12.2?

### ISSUE THREAD

**Issue by hustlmh** [ByteDance-Seed/Bagel#77]
Hi, I’m using several NVIDIA TITAN X (Pascal) GPU with NVIDIA driver 535.230.02, which only supports up to CUDA 12.2. I can’t upgrade to CUDA 12.4 due to hardware limitations.

I want to run the BAGEL model, which depends on flash_attention. However:

PyTorch 2.5.1 + TorchVision 0.20.1 requires CUDA 12.4 (not compatible).
I have a precompiled flash_attn-2.5.8 wheel for CUDA 12.2 and PyTorch 2.3 (cu122torch2.3cxx11abiFALSE), and I plan to use PyTorch 2.3.1 + TorchVision 0.18.1.
Questions:

1）Is it feasible to run BAGEL with PyTorch 2.3.1 and TorchVision 0.18.1? Are there known compatibility issues?
2）Will the flash_attn-2.5.8 wheel work with this setup?
3）Do I need to adjust code or configurations (e.g., environment variables, dependencies)?
Any advice would be greatly appreciated! Thanks.

**Comment by Andy1621** [comment]
Hi! For training, we use PyTorch 2.5.0+ because of `FlexAttention`.
For inference, I think you can try older versions since we use `FlashAttention`. And the code does not need to be changed.

**Comment by zhujingze** [comment]
请问你这些版本成功配好环境了吗？
